src/Monitor.ts

```diff
@@ -11,94 +11,64 @@ interface Config {
 }
 
 class Monitor {
-    private statsdClient: AsyncStatsD | undefined;
-    private config: Config;
+    private plugins: PluginsWrapper;
     private logger: Logger;
+    private config: Config;
 
     constructor(options: MonitorOptions) {
-        this.config = {
-            serviceName: options.serviceName ? `${options.serviceName}.` : '',
-            shouldMonitorExecutionStart: options.shouldMonitorExecutionStart ?? true,
-            disableSuccessLogs: options.logging?.disableSuccessLogs ?? false,
-        };
-
         this.logger = new Logger(options.mock ? emptyLogger : options.logging?.logger ?? consoleLogger, {
             logErrorsAsWarnings: options.logging?.logErrorsAsWarnings,
         });
 
-        if (options.statsd) {
-            const {apiKey, root, port, ...restStatsdOptions} = options.statsd;
-            const prefixesArray = [apiKey, root, options.serviceName].filter(Boolean);
-            const prefix = `${prefixesArray.join('.')}.`;
+        this.plugins = new PluginsWrapper(options.plugins, {
+            logger: this.logger,
+        });
 
-            this.statsdClient = new AsyncStatsD(this.logger, {
-                port: port ?? 8125,
-                prefix,
-                mock: options.mock,
-                ...restStatsdOptions,
-            });
-        }
+        this.config = {
+            serviceName: options.serviceName ? `${options.serviceName}.` : '',
```

Comment:
why the . in the end of the service name?

Reply:
Something tells me this is statsd leftovers that should be moved to the plugin. I'll look at it to validate

Reply:
https://github.com/Soluto/monitored/pull/15/files#diff-9a8b0119a549f495988da11d319132f85827df423b8d1e58a4859f01094b71eaR31

This is statsd leftovers. Doesn't look like `serviceName` is used inside of `Monitor` anyway.